LinuxPasswordManager error on Archlinux

Downloaded the 0.1.0 release. I use Archlinux. I'm going to assume I'm missing a package but I don't know which one (and maybe it can also handle this more gracefully?)

```
[bbq@luna bohning]$ chmod +x USDB_Syncer-0.1.0-Linux 
[bbq@luna bohning]$ ./USDB_Syncer-0.1.0-Linux 
Gtk-Message: 14:27:02.969: Failed to load module "canberra-gtk-module"
Gtk-Message: 14:27:02.970: Failed to load module "canberra-gtk-module"

(USDB_Syncer-0.1.0-Linux:3831): dbind-WARNING **: 14:27:02.976: AT-SPI: Error retrieving accessibility bus address: org.a11y.Bus.Error: Failed to execute child process “/usr/bin/dbus-broker-launch” (No such file or directory)

<removed a whole bunch of GTK theme warnings since a lot of programs do that>

2023-05-13 14:27:03 [ERROR] Uncaught exception:
    Traceback (most recent call last):
      File "usdb_syncer/main.py", line 36, in <module>
      File "usdb_syncer/main.py", line 31, in cli_entry
      File "usdb_syncer/gui/mw.py", line 309, in main
      File "usdb_syncer/gui/mw.py", line 318, in _load_main_window
      File "usdb_syncer/song_list_fetcher.py", line 23, in get_all_song_data
      File "usdb_syncer/song_list_fetcher.py", line 49, in get_available_songs
      File "usdb_syncer/usdb_scraper.py", line 333, in get_usdb_available_songs
      File "usdb_syncer/usdb_scraper.py", line 231, in get_usdb_page
      File "usdb_syncer/usdb_scraper.py", line 222, in page
      File "usdb_syncer/usdb_scraper.py", line 78, in session
      File "usdb_syncer/usdb_scraper.py", line 64, in new_session_with_cookies
      File "usdb_syncer/settings.py", line 224, in cookies
      File "browser_cookie3/__init__.py", line 973, in chrome
      File "browser_cookie3/__init__.py", line 516, in __init__
      File "browser_cookie3/__init__.py", line 331, in __init__
      File "browser_cookie3/__init__.py", line 344, in __add_key_and_cookie_file
      File "browser_cookie3/__init__.py", line 238, in get_password
    TypeError: _LinuxPasswordManager.__get_kdewallet_password_jeepney() missing 1 required positional argument: 'key'
[3831] Failed to execute script 'main' due to unhandled exception!
```

@mjhalwa this is definitely a big improvement!

```
[bbq@luna bohning]$ ./USDB_Syncer-0.1.18-Linux 
Gtk-Message: 16:03:25.838: Failed to load module "canberra-gtk-module"
Gtk-Message: 16:03:25.838: Failed to load module "canberra-gtk-module"

(USDB_Syncer-0.1.18-Linux:4036): dbind-WARNING **: 16:03:25.844: AT-SPI: Error retrieving accessibility bus address: org.a11y.Bus.Error: Failed to execute child process “/usr/bin/dbus-broker-launch” (No such file or directory)

<removed a whole bunch of GTK theme warnings since a lot of programs do that>

2023-05-26 16:03:25 [DEBUG] Failed to retrieve Chrome cookies.
2023-05-26 16:03:25 [DEBUG] Starting new HTTPS connection (1): usdb.animux.de:443
2023-05-26 16:03:26 [DEBUG] https://usdb.animux.de:443 "GET /?link=profil HTTP/1.1" 200 None
2023-05-26 16:03:26 [DEBUG] Loading KWallet
2023-05-26 16:03:26 [DEBUG] Loading SecretService
2023-05-26 16:03:26 [DEBUG] Loading Windows
2023-05-26 16:03:26 [DEBUG] Loading chainer
2023-05-26 16:03:26 [DEBUG] Loading libsecret
2023-05-26 16:03:26 [DEBUG] Loading macOS
2023-05-26 16:03:26 [ERROR] Uncaught exception:
    Traceback (most recent call last):
      File "usdb_syncer/main.py", line 36, in <module>
      File "usdb_syncer/main.py", line 31, in cli_entry
      File "usdb_syncer/gui/mw.py", line 399, in main
      File "usdb_syncer/gui/mw.py", line 408, in _load_main_window
      File "usdb_syncer/song_list_fetcher.py", line 23, in get_all_song_data
      File "usdb_syncer/song_list_fetcher.py", line 49, in get_available_songs
      File "usdb_syncer/usdb_scraper.py", line 333, in get_usdb_available_songs
      File "usdb_syncer/usdb_scraper.py", line 231, in get_usdb_page
      File "usdb_syncer/usdb_scraper.py", line 222, in page
      File "usdb_syncer/usdb_scraper.py", line 79, in session
      File "usdb_syncer/usdb_scraper.py", line 49, in establish_usdb_login
      File "usdb_syncer/settings.py", line 31, in get_usdb_auth
      File "keyring/core.py", line 55, in get_password
      File "keyring/backends/fail.py", line 25, in get_password
    keyring.errors.NoKeyringError: No recommended backend was available. Install a recommended 3rd party backend package; or, install the keyrings.alt package if you want to use the non-recommended backends. See https://pypi.org/project/keyring for details.
[4036] Failed to execute script 'main' due to unhandled exception!
```

The `NoKeyringError` also contains a link to https://pypi.org/project/keyring, which is already tons more useful [compared to the KDE error]. I assume it is not planned to ever make USDB Syncer work without any kind of keyring installed? I'll try and install one regardless and see if maybe they've improved from the session-destroying things they were way back in the day, because unless a plaintext alternative is also in the works, it actually makes sense for USDB Syncer to require one.

If you want to cater to that one Linux user that _doesn't_ know what a terminal is, you might consider catching the `NoKeyringError` and putting the message in a UI box (because right now it's just the splash for a second and then it disappears) but that should probably be its own ticket, and I'm not too fussed about it.

--- a/.github/workflows/release.yaml
+++ b/.github/workflows/release.yaml
@@ -23,6 +23,11 @@ jobs:
             PYINSTALLER_ARGS: "--onefile --icon src/usdb_syncer/gui/resources/appicon_128x128.png --add-data 'data/song_list.json;data'"
     steps:
       - uses: actions/checkout@v3
+      - name: Install Linux specific dependencies for dbus-python
+        if: matrix.TARGET == 'Linux'
+        run: |
+          sudo apt install -y gcc python3-dev libdbus-1-dev
+          pkg-config --cflags --libs dbus-1
       - name: Install Qt
         uses: jurplel/install-qt-action@v3
         with:

--- a/.github/workflows/tox.yaml
+++ b/.github/workflows/tox.yaml
@@ -15,6 +15,10 @@ jobs:
 
     steps:
       - uses: actions/checkout@v3
+      - name: Install Linux specific dependencies for dbus-python
+        run: |
+          sudo apt install -y gcc python3-dev libdbus-1-dev
+          pkg-config --cflags --libs dbus-1
       - name: Set up Python ${{ matrix.python-version }}
         uses: actions/setup-python@v4
         with:

--- a/README.md
+++ b/README.md
@@ -6,6 +6,15 @@
 [![Release](https://github.com/bohning/usdb_syncer/actions/workflows/release.yaml/badge.svg)](https://github.com/bohning/usdb_syncer/actions/workflows/release.yaml)
 [![tox](https://github.com/bohning/usdb_syncer/actions/workflows/tox.yaml/badge.svg)](https://github.com/bohning/usdb_syncer/actions/workflows/tox.yaml)
 
+## Install (Linux)
+
+requires extra packages on Linux
+
+``` bash
+apt update
+apt install libdbus-1-3
+```
+
 ## Development
 
 **USDB Syncer** is written in Python.
@@ -13,6 +22,13 @@ The following explains how to set up a development environment.
 A Python 3.11 installation is assumed.
 
 ### Python Setup
+
+requires extra packages when developing **on Linux**
+
+``` bash
+apt install -y gcc python3-dev libdbus-1-dev
+pkg-config --cflags --libs dbus-1
+```
 
 [tox](https://github.com/tox-dev/tox) makes it easy to run the full CI pipeline on your local machine, i.e., if the pipeline passes on your machine there is a good chance it will also pass on the build server.
 The minimal setup to develop your package is to create a virtual environment and install the package and its runtime requirements:

--- a/README.md
+++ b/README.md
@@ -15,6 +15,14 @@ apt update
 apt install libdbus-1-3
 ```
 
+### Troubleshooting
+
+- The `keyring` package auto-detects an appropriate installed keyring backend (see [PyPI - keyring](https://pypi.org/project/keyring/)). Thus may require following additional package if no backend can be detected, see #136
+
+  ``` bash
+  apt install gnome-keyring
+  ```
+
 ## Development
 
 **USDB Syncer** is written in Python.

--- a/setup.cfg
+++ b/setup.cfg
@@ -33,6 +33,7 @@ install_requires =
     mutagen
     iso639-lang
     keyring
+    dbus-python;sys_platform=="linux"
 
 [options.packages.find]
 where = src

